File: app/servicos/sei.py
"""Interface com o SEI - DEFINIDA E VAZIA, de proposito.

A v1 NAO integra com o SEI: nenhuma API, nenhum scraping. O numero do processo
e digitado (validado em app/servicos/nup.py) e `numero_documento_sei` /
`url_permanente` sao campos manuais.

Este modulo existe para que, quando a integracao for autorizada, exista um unico
ponto de acoplamento - e para deixar explicito que hoje nada aqui esta ligado.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Protocol

VERSAO_SEI_ALVO = "4.0.12.15"
BASE_SEI = "sei.ufvjm.edu.br"
UNIDADE_SEI = "csso.sisa"

PASSOS_INCLUSAO = (
    "No SEI, abra o processo pelo NUP.",
    "Incluir Documento -> Externo.",
    "Tipo do Documento: Parecer Técnico.",
    "Anexe o PDF gerado pelo sistema e assine no SEI.",
    "Volte aqui e preencha o nº do documento SEI, a data e o link permanente.",
)


class NaoImplementadoNaV1(NotImplementedError):
    """Integracao com o SEI esta fora do escopo da v1 (SS1 do prompt)."""


@dataclass(frozen=True)
class DocumentoSei:
    numero: str
    url_permanente: str | None = None


class ClienteSei(Protocol):
    """Contrato que uma futura integracao devera cumprir."""

    def incluir_documento_externo(
        self, nup: str, caminho_pdf: str, tipo: str
    ) -> DocumentoSei: ...

    def consultar_processo(self, nup: str) -> dict: ...


class ClienteSeiIndisponivel:
    """Implementacao padrao: recusa qualquer chamada, com mensagem clara."""

    def incluir_documento_externo(self, nup: str, caminho_pdf: str, tipo: str):
        raise NaoImplementadoNaV1(
            "A v1 nao integra com o SEI. Inclua o documento manualmente: "
            + " ".join(PASSOS_INCLUSAO)
        )

    def consultar_processo(self, nup: str) -> dict:
        raise NaoImplementadoNaV1("A v1 nao consulta o SEI. Digite o NUP.")


def obter_cliente() -> ClienteSei:
    return ClienteSeiIndisponivel()

File: nuvem/src/servicos/sei.ts
/**
 * Interface com o SEI — DEFINIDA E VAZIA, de propósito.
 * Porte de `app/servicos/sei.py`.
 *
 * A v1 NÃO integra com o SEI: nenhuma API, nenhum scraping. O número do
 * processo é digitado (validado em `nup.ts`) e `numero_documento_sei` /
 * `url_permanente` são campos manuais.
 *
 * Este módulo existe para que, quando a integração for autorizada, exista um
 * único ponto de acoplamento — e para deixar explícito que hoje nada aqui está
 * ligado.
 */

export const VERSAO_SEI_ALVO = "4.0.12.15";
export const BASE_SEI = "sei.ufvjm.edu.br";
export const UNIDADE_SEI = "csso.sisa";

export const PASSOS_INCLUSAO: readonly string[] = Object.freeze([
  "No SEI, abra o processo pelo NUP.",
  "Incluir Documento -> Externo.",
  "Tipo do Documento: Parecer Técnico.",
  "Anexe o PDF gerado pelo sistema e assine no SEI.",
  "Volte aqui e preencha o nº do documento SEI, a data e o link permanente.",
]);

/** Integração com o SEI está fora do escopo da v1 (§1 do prompt). */
export class NaoImplementadoNaV1 extends Error {}

export interface DocumentoSei {
  readonly numero: string;
  readonly url_permanente: string | null;
}

/** Contrato que uma futura integração deverá cumprir. */
export interface ClienteSei {
  incluir_documento_externo(nup: string, caminho_pdf: string, tipo: string): DocumentoSei;
  consultar_processo(nup: string): Record<string, unknown>;
}

/** Implementação padrão: recusa qualquer chamada, com mensagem clara. */
export class ClienteSeiIndisponivel implements ClienteSei {
  incluir_documento_externo(_nup: string, _caminho_pdf: string, _tipo: string): DocumentoSei {
    throw new NaoImplementadoNaV1(
      "A v1 nao integra com o SEI. Inclua o documento manualmente: " + PASSOS_INCLUSAO.join(" "),
    );
  }

  consultar_processo(_nup: string): Record<string, unknown> {
    throw new NaoImplementadoNaV1("A v1 nao consulta o SEI. Digite o NUP.");
  }
}

export function obter_cliente(): ClienteSei {
  return new ClienteSeiIndisponivel();
}
